File: backend/models/task.go
package models

import (
	"gorm.io/gorm"
)

type Task struct {
	gorm.Model
	Title     string `json:"title"`
	Details   string `json:"details"`
	DueDate   string `json:"due_date"`
	Completed bool   `json:"completed"`
	DealID    uint   `json:"deal_id"`
	Deal      Deal   `json:"deal" gorm:"foreignKey:DealID"`
	UserID    uint   `json:"user_id"`
	User      User   `json:"user" gorm:"foreignKey:UserID"`
}

File: frontend/app/src/types/Task.ts
import { User } from './User';
import { Deal } from './Deal';

export interface Task {
  id: number;
  title: string;
  details: string;
  due_date: string;
  completed: boolean;
  deal_id: number;
  user_id: number;
  created_at: string;
  updated_at: string;
  deal?: Deal;
  user?: User;
}

export interface TaskFormData {
  title: string;
  details: string;
  due_date: string;
  deal_id?: number;
}
